src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -26,13 +30,180 @@
+    }
+
+    $field_wrapper = Html::getClass($this->fieldDefinition->getName()) . '-add-another-group';
+
+    $elements = [
+      '#type' => 'container',
+      '#tree' => TRUE,
+      '#title' => $this->t('Other groups'),
+      '#description' => $description,
+      '#prefix' => '<div id="' . $field_wrapper . '">',
+      '#suffix' => '</div>',
+      '#cardinality' => FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED,
+      '#cardinality_multiple' => TRUE,
+      '#theme' => 'field_multiple_value_form',
+      '#field_name' => $this->fieldDefinition->getName(),
+      '#max_delta' => 1,
+    ];
+
+    $elements['add_more'] = [
+      '#type' => 'button',
+      '#value' => $this->t('Add another item'),
+      '#name' => 'add_another_group',
+      '#ajax' => [
+        'callback' => [$this, 'addMoreAjax'],
+        'wrapper' => $field_wrapper,
+        'effect' => 'fade',
+      ],
+    ];
+
+    $start_key = 0;
+
+    $target_type = $this->fieldDefinition->getTargetEntityTypeId();
+    $entity_group_ids = array_map(function($entity) {
+      return $entity->id();
+    }, $items->referencedEntities());
+
+    $other_groups = Og::getEntityGroups('user', \Drupal::currentUser()->id());
+
+    foreach ($other_groups[$target_type] as $other_group) {
+      if (!in_array($other_group->id(), $entity_group_ids)) {
+        continue;
+      }
+      $elements[$start_key] = $this->otherGroupsSingle($start_key, $other_group);
+      $start_key++;
+    }
+
+    if (!$form_state->get('other_group_delta')) {
+      $form_state->set('other_group_delta', $start_key);
+    }
+
+    // Get the trigger element and check if this the add another item button.
+    $trigger_element = $form_state->getTriggeringElement();
+
+    if ( $trigger_element['#name'] == 'add_another_group') {
+      // Increase the number of other groups.
+      $delta = $form_state->get('other_group_delta') + 1;
+      $form_state->set('other_group_delta', $delta);
+    }
+
+    // Add another auto complete field.
+    for ($i = $start_key; $i <= $form_state->get('other_group_delta'); $i++) {
+      $elements[$i] = $this->otherGroupsSingle($i);
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Generating other groups auto complete element.
+   *
+   * @param $delta
+   *   The delta of the new element. Need to be the last delta in order to be
+   *   added in the end of the list.
+   * @param EntityInterface|NULL $entity
+   *   The entity object.
+   * @return array
+   *   A single entity reference input.
+   */
+  public function otherGroupsSingle($delta, EntityInterface $entity = NULL) {
+    return [
+      'target_id' => [
+        '#type' => 'entity_autocomplete',
+        '#target_type' => $this->fieldDefinition->getTargetEntityTypeId(),
+        // todo: fix the definition in th UI level.
+        '#selection_handler' => 'default:og',
+        '#selection_settings' => [
+          'other_groups' => TRUE,
+          'field_mode' => 'admin',
```

Comment:
@amitaibu Here. This is where we pass the admin field mode. Related to field modes, if we want to keep selection handlers knowing less, we need to pass this in the selection settings like this. The selection handler still has to know about the current user. Hopefully we an fix that later on so it just indirectly relies on another service.